src/http_client_async.zig

```diff
@@ -2054,14 +2124,52 @@ pub fn onData(this: *HTTPClient, comptime is_ssl: bool, incoming_data: []const u
     }
 }
 
-fn fail(this: *HTTPClient, err: anyerror) void {
+pub fn closeAndAbort(this: *HTTPClient, reason: JSC.JSValue, comptime is_ssl: bool, socket: NewHTTPContext(is_ssl).HTTPSocket) void {
+    socket.ext(**anyopaque).?.* = bun.cast(
+        **anyopaque,
+        NewHTTPContext(is_ssl).ActiveSocket.init(&dead_socket).ptr(),
+    );
+    if (this.globalThis) |globalThis| {
```

Comment:
I don't think it's threadsafe to create JSValue here or for it to be aware of globalThis

I think it will need to enqueue a task to the event loop and to do this on the main thread